### PR TITLE
retain width and height after resize for master

### PR DESCRIPTION
**Pull request by tonylitianyu**
Reopen a pull request for the master branch. The width and height were modified after this resize call. Now the width and height will be retained if they were set before. A new method is also provided to input the set width and height.

Signed-off-by: tonylitianyu <tonylitianyu@gmail.com>

### CODE REVIEW

File: sensor_msgs/include/sensor_msgs/point_cloud2_iterator.hpp
Comment (jacobperron) [inline sensor_msgs/include/sensor_msgs/point_cloud2_iterator.hpp:129]: This should be separated into two params:

```suggestion
   * Updates the size and sets the width and height of the original sensors_msgs::msg::PointCloud2.
   * @param width The new width of the point cloud.
   * @param height The new height of the point cloud.
```

**Review by jacobperron — COMMENTED** [review]
Thanks for the pull request!
Could you also add test to [this file](https://github.com/ros2/common_interfaces/blob/master/sensor_msgs/test/test_pointcloud_iterator.cpp) for the resize methods?

File: sensor_msgs/include/sensor_msgs/point_cloud2_iterator.hpp
Comment (jacobperron) [inline sensor_msgs/include/sensor_msgs/point_cloud2_iterator.hpp:130]:
```suggestion
   * @param height The new height of the point cloud.
```

File: sensor_msgs/test/test_pointcloud_iterator.cpp
Comment (jacobperron) [inline sensor_msgs/test/test_pointcloud_iterator.cpp:138]: Prefer using `static_cast` and `uint32_t` (which is the actual type of the width and height members).

It might also be fine to just declare the literals as unsigned, like `1ul`.

**Review by jacobperron — COMMENTED** [review]
Tests look good to me, thanks for adding them.
I have a few minor comments; after those are addressed I we should trigger CI before merging.

**Review by jacobperron — APPROVED** [review]
LGTM pending CI

**Comment by tonylitianyu** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16768/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11349/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17166/)

**Review by jacobperron — APPROVED** [review]
Thanks for the fix!

**Comment by Blast545** [comment]
:man_farmer: It seems this PR introduced some test regressions on the windows debug buildfarm jobs:
https://ci.ros2.org/view/nightly/job/nightly_win_deb/2381/msbuild/folder.-1615888395/

I think in the resize function, instead of using `size_t`, you should be using `uint32_t`. Can I ask you to take a look? @tonylitianyu

**Comment by tonylitianyu** [comment]
new PR added for the type issue
https://github.com/ros2/common_interfaces/pull/194

**Pull request merged**
